files/__addonLocation__/types/glint.d.ts

```diff
@@ -0,0 +1,10 @@
+import '@glint/environment-ember-loose';
```

Comment:
should this have been committed?

Reply:
Yes, this is intentional. At least that's what worked well for me.

So *this* file is for making Glint aware of what your own addon *consumes*. Say this addon depends on your `ember-popperjs`. With classic templates, invoking `<PopperJS>` would make Glint fail, as it doesn't know what `PopperJS` is. So you would import the component here, and add its type to the registry. 

The other file `src/glint.ts` is about making Glint aware of what your addon *exposes*. So the app would import this file (which is published on npm!), and have all the components etc. added to the app's registry. 

This is what we came up with here: https://github.com/typed-ember/glint/discussions/303. And which is documented here: https://typed-ember.gitbook.io/glint/using-glint/ember/authoring-addons#publishing-addons-with-glint-types

Reply:
should we link to that discussion in this `glint.d.ts` file?
it feels "non-standard" to me, but idk if that's because Glint is new.
even so, it's a _hidden_ convention (no tooling/linting, to enforce, etc) :thinking:

Reply:
I think this relates to the question above about how we want to handle addon-only unpublished type declarations and making it as clear as possible to authors/readers the role they play.

Reply:
I'm now 100% in camp "TS addons should have a types directory for local unpublished types"